src/js/lazy-function.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Make function application lazy.
+ *
+ * @param {function} func - The function to be executed lazily.
+ * @returns {function(...[*]=): *}
+ */
+function lazyFunction(func) {
```

Comment:
is it really needed or do we just want to play with coolness here?

Reply:
remember, a very good code is very simple to read.

Reply:
i just fear a bit that this creates not easy to read code.

Reply:
Yeah it's need for prop-types to properly check trees

Reply:
ok then :)

Reply:
might be better, its up too u